### PR TITLE
Jcsp 708 add csv board usages reading

### PR DESCRIPTION
**Pull request by anastasiiagnes**
Adding basic functionality for reading the most frequently used boards from a csv file and opening these boards pages in scope of the test.

### CODE REVIEW

File: custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.net.URI
+
+object CsvBoardsReader {
+
+    var boardsList: MutableList<CsvBoard>?
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+    private const val csvPropertyFile = "BoardUsageFrequency.csv"
+
+    init {
+        boardsList = getBoardsFromCsv()
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt]: Let's move the list in BoardsFrequencyManager and count totalWeight only once

**Review by ysemenkov — CHANGES_REQUESTED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt (hunk 1 above)
Comment (anastasiiagnes) [reply custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt]: Ok, got you, will fix this.

File: custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.net.URI
+
+object CsvBoardsReader {
+
+    var csvContent: MutableList<CsvBoard>?
+    private val logger: Logger = LogManager.getLogger(this::class.java)
+    private const val csvPropertyFile = "BoardUsageFrequency.csv"
+
+    init {
+        csvContent = readBoardsFromCsv()
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt]: csvContent  is redundant here
this is a reader but not a holder

Let's just use readBoardsFromCsv(csvPropertyFile: String) in BoardsFrequencyManager

**Review by ysemenkov — CHANGES_REQUESTED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/utils/boards/BoardsFrequencyManager.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package jces1209.vu.utils.boards
+
+import java.util.*
+
+class BoardsFrequencyManager {
+
+    private val csvPropertyFile = "BoardUsageFrequency.csv"
```
Comment (ysemenkov) [inline custom-vu/src/main/kotlin/jces1209/vu/utils/boards/BoardsFrequencyManager.kt:16]: Please add a comment with the description regarding this file: `location`, `template`, etc
and add a TODO to get the file path from the tenant properties

**Review by ysemenkov — CHANGES_REQUESTED** [review]

**Review by NataliiaShavshyna — DISMISSED** [review]

**Review by ysemenkov — APPROVED** [review]

File: custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt (hunk 2 above)
Comment (anastasiiagnes) [reply custom-vu/src/main/kotlin/jces1209/vu/utils/boards/CsvBoardsReader.kt]: Fixed according to the comment.

**Pull request merged**